src/FakeItEasy/Creation/CastleDynamicProxy/CastleDynamicProxyGenerator.cs

```diff
@@ -19,13 +21,15 @@ internal class CastleDynamicProxyGenerator
         private static readonly ProxyGenerator ProxyGenerator = new ProxyGenerator();
         private readonly CastleDynamicProxyInterceptionValidator interceptionValidator;
 
+#if FEATURE_SECURITY_PERMISSIONS
         [SuppressMessage("Microsoft.Performance", "CA1810:InitializeReferenceTypeStaticFieldsInline", Justification = "No field initialization.")]
         static CastleDynamicProxyGenerator()
         {
 #pragma warning disable 618
             AttributesToAvoidReplicating.Add(typeof(SecurityPermissionAttribute));
 #pragma warning restore 618
         }
+#endif

```

Comment:
@blairconrad yes, you should be able to drop the static constructor for all frameworks.